lib/services/watcher_test.go

```diff
@@ -1094,10 +1094,11 @@ func TestSAMLIdPServiceProviderWatcher(t *testing.T) {
 
 	// The first event is always the current list of service providers.
 	select {
-	case changeset := <-w.SAMLIdPServiceProvidersC:
-		require.Len(t, changeset, 1)
-		require.Empty(t, resourceDiff(changeset[0], sp1))
-	case <-w.Done():
+	case changeset := <-w.Collector.Cfg.SAMLIdPServiceProvidersC:
+		require.Empty(t, cmp.Diff(types.SAMLIdPServiceProviders{sp1}, changeset,
+			cmpopts.IgnoreFields(types.Metadata{}, "ID"),
+		), "should be no differences in the changeset after adding the first service provider")
```

Comment:
nit: make indent levels clearer?

```suggestion
		require.Empty(t,
			cmp.Diff(
				types.SAMLIdPServiceProviders{sp1},
				changeset,
				cmpopts.IgnoreFields(types.Metadata{}, "ID")),
			"should be no differences in the changeset after adding the first service provider")
```